### PR TITLE
kafka_sender checks DataEntity metadata _key 

### PR DESCRIPTION
**Pull request by ciorg**
* kafka_sender defaults to check for DataEntity metadat _key for id_field
* updated tests
* updated readme

### CODE REVIEW

File: asset/src/kafka_sender/processor.ts [hunk 1]
```diff
@@ -45,10 +45,10 @@ export default class KafkaSender extends BatchProcessor<KafkaSenderConfig> {
     }
 
     private getKey(msg: DataEntity): string|null {
-        if (!this.opConfig.id_field) return null;
+        const key = msg[this.opConfig.id_field] || msg.getMetadata('_key');
 
-        const key = msg[this.opConfig.id_field];
-        if (key && isString(key)) return key;
+        if (key === undefined) return null;
```
Comment (kstaken) [inline asset/src/kafka_sender/processor.ts]: Is this actually necessary? Wouldn't undefined achieve the same result?

Comment (ciorg) [reply asset/src/kafka_sender/processor.ts]: not sure, but the tests fail if undefined is returned.

Comment (kstaken) [reply asset/src/kafka_sender/processor.ts]: OK, just consider peter's feedback.

**Comment by ciorg** [comment]
What is the issue if the opConfig.id_field and metadata response are treated the same?  Doesn't it just need to return a string?  Could the metadata _key be a non-string?

Is there an issue I'm not considering by using `const key = msg[this.opConfig.id_field] || msg.getMetadata('_key');`

**Comment by peterdemartini** [comment]
> What is the issue if the opConfig.id_field and metadata response are treated the same? Doesn't it just need to return a string? Could the metadata _key be a non-string?
> 
> Is there an issue I'm not considering by using `const key = msg[this.opConfig.id_field] || msg.getMetadata('_key');`

@ciorg the main issue the validation, the only time it should throw an error is when the `id_field` is set.

**Pull request merged**